src/features/renameVariableParts.ts

```diff
@@ -81,18 +87,19 @@ export const registerRenameVariableParts = () => {
         )
         quickPick.onDidAccept(() => {
             const activeItem = quickPick.activeItems[0]!
-            if (editingIndex === undefined) {
-                editingIndex = quickPick.items.indexOf(activeItem)
-                const { label } = activeItem
-                quickPick.items = [
-                    /* { label } */
-                ]
-                quickPick.value = label
-            } else {
+            if (editingIndex) {
```

Comment:
@ILScc for now, CI is failing (spoiler because of prettier). Also I love refactors of any scale, but this one introduced a bug. As you can see it was checked against undefined on line above as index can be `0`, which is common case.
Because of this change when, editing index 0, user need to double press enter before going out of the so called edit mode (because it would set to -1 on first call). I recommend using [P42](https://p42.ai/documentation/code-assist/#branching) for this:
1. [Invert condition](https://p42.ai/documentation/code-assist/invert-condition) (or Flip if/else from abrocadabra)
2. [Remove unnecessary else](https://p42.ai/documentation/code-assist/remove-unnecessary-else)

Note that all these tools also support Vue.

Of course I could've used linter rule here, but I don't it to be annoying as I experienced a lot of cases, where I couldn't use it because of readability. But it was a good decision in this case.

Reply:
Thank for explanation. Fixed it in [679998cb2340b7151863e9477241dfa4c98beb68](https://github.com/zardoy/vscode-experiments/pull/4/commits/679998cb2340b7151863e9477241dfa4c98beb68)